### ISSUE TITLE
Decrease allocations done

### ISSUE THREAD

**Issue by Datseris** [JuliaDynamics/DynamicalBilliards.jl#35]
A standard evolve call has at least 100,000 allocations  even though everything is done through SVectors.

I do not know how or why this happens and also how to solve it.

Just do 
```julia
using DynamicalBilliards, BenchmarkTools
bt = billiard_sinai()
p = randominside(bt)
evolve!(p, bt, 1.0) # compile
@btime evolve!(p, bt, 1000.0)
```
to see what I mean.

**Comment by dapias** [comment]
What kind of billiard `bt` is?

**Comment by Datseris** [comment]
It doesn't really matter. The allocations come from `evolve!` so I am quite sure they happen in *any* combination of billiard tables.

And so it should be, since all the methods are identical no matter the obstacles. What changes is only sings and numerical values...

**Comment by Datseris** [comment]
I guess the answer is simply "SVectors result in allocations" ? I cannot know for sure though.

What I know for sure is that **a lot of** SVectors are created all the time during `evolve!`

**Comment by dapias** [comment]
Check out this case:
```julia
using DynamicalBilliards, BenchmarkTools

space = 3.0
r = 1.0
polygon_sides = 6
bth = billiard_polygon(polygon_sides, space/sqrt(3); setting = "periodic")
d = Disk([0.,0.], r)
push!(bth, d)
t = 1000.0
p = randominside(bth)
@btime evolve!(p, bth, t)
```
I get
```
478.572 μs (7459 allocations: 137.20 KiB)
```
PS: Sometimes in Sydney, the connection to github is down. I wanted to share this piece of code two days ago...

**Comment by dpsanders** [comment]
Note that when using the benchmarking tools, any global variables should be preceded by a `$`, as e.g.
`@btime evolve!($p, $bth, $t)`.
However, I don't *think* this will make much difference here.

**Comment by dapias** [comment]
Oh ok, thank you @dpsanders

**Comment by Datseris** [comment]
This is apparently due to the type instabilities that result from using `Vector{Obstacle{T}}` as the billiard table.

I am hoping the combo of #30 and #51 will resolve it at least partly.

**Comment by Datseris** [comment]
Closed in #72
